### ISSUE TITLE
Add protobuf DESCRIPTOR objects as attributes of generated classes and methods

### ISSUE THREAD

**Issue by nathanielmanistaatgoogle** [grpc/grpc#11657]
The generated `_pb2_grpc.py` code should look something like
```python
class MyServiceStub(object):
  DESCRIPTOR = my_service_pb2.DESCRIPTOR.services_by_name['MyService']

  def __init__(self, channel):
    self.MyMethod = channel.unary_unary(
        '/my_package.MyService/MyMethod',
       request_serializer=my_messages__pb2.MyMethodRequest.SerializeToString,
       response_deserializer=my_messages__pb2.MyMethodResponse.FromString,
       descriptor=my_service_pb2.DESCRIPTOR.services_by_name['MyService'].methods_by_name['MyMethod'],
      )
```
so that
```python
my_stub = my_service_pb2_grpc.MyServiceStub(my_channel)
my_service_descriptor = my_stub.DESCRIPTOR
my_method_descriptor = my_stub.MyMethod.DESCRIPTOR
```
works and is useful code.

**Comment by amauryfa** [comment]
The call to `channel.unary_unary` could also use the new descriptor:
  `descriptor=self.DESCRIPTOR.methods_by_name['MyMethod']`

**Comment by nathanielmanistaatgoogle** [comment]
@amauryfa: it could, but should it? Do we want to support applications subclassing the stub class (well, yes!), shadowing the class-scope `DESCRIPTOR` constant (well... I don't know...), and observing their shadowing value being used in `channel.unary_unary` calls made during instantiation of their stub subclass (this part seems really crazy and/or sketchy)? What would be the use case? What would be the motivation for us to provide (and support!) this application-exposed behavior?

**Comment by amauryfa** [comment]
Applications which subclass and mess with public documented symbols are on their own. This is true for any Python object.

**Comment by nathanielmanistaatgoogle** [comment]
I started work on this but I think this is currently blocked on [Protocol Buffers issue 3452](https://github.com/google/protobuf/issues/3452).

**Comment by nathanielmanistaatgoogle** [comment]
I'm going to call this a duplicate of [issue 3898](https://github.com/grpc/grpc/issues/3898); closing.
